enterprise/internal/codeintel/context/service.go

```diff
@@ -2,26 +2,770 @@ package context
+
+// 			defsList = append(defsList, defs...)
+// 		}
+// 	}
+
+// 	clippedContent := map[string]struct{}{}
+// 	// var syntectDocsList []*scip.Document
+// 	for _, def := range defsList {
+// 		file, err := s.gitserverClient.ReadFile(
+// 			ctx,
+// 			authz.DefaultSubRepoPermsChecker,
+// 			api.RepoName(def.Dump.RepositoryName),
+// 			api.CommitID(def.Dump.Commit),
+// 			def.Path,
+// 		)
+// 		if err != nil {
+// 			return "", err
+// 		}
+// 		c := strings.Split(string(file), "\n")
+// 		// fmt.Println("HERE IS THE c", c)
+
+// 		syntectDocs, err := s.getSCIPDocumentByContent(ctx, string(file), def.Path)
+// 		if err != nil {
+// 			return "", err
+// 		}
+
+// 		prd := precise.RangeData{
+// 			StartLine:      def.TargetRange.Start.Line,
+// 			EndLine:        def.TargetRange.End.Line,
+// 			StartCharacter: def.TargetRange.Start.Character,
+// 			EndCharacter:   def.TargetRange.End.Character,
+// 		}
+
+// 		for _, occ := range syntectDocs.Occurrences {
+// 			r := scip.NewRange(occ.EnclosingRange)
+// 			if precise.RangeIntersectsSpan(prd, int(r.Start.Line), int(r.End.Line)) {
+// 				// z := strings.Join(c[r.Start.Line:r.End.Line], "\n")
+// 				// zz := z[r.Start.Character:r.End.Character]
+// 				// fmt.Println("HERE IS THE zz \n", zz)
+// 				// fmt.Println("HERE IS THE r.Start.Line", r.Start.Line)
+// 				// fmt.Println("HERE IS THE r.End.Line", r.End.Line)
+// 				// fmt.Println("HERE IS THE r.Start.Character", r.Start.Character)
+// 				// fmt.Println("HERE IS THE r.End.Character", r.End.Character)
+// 				snpt := extractSnippet(c, r.Start.Line, r.End.Line, r.Start.Character, r.End.Character)
+// 				// fmt.Println("HERE IS THE SNIPPETS", snpt)
+// 				// cnt := pageContent(c, &r.Start.Line, &r.End.Line)
+// 				// clippedContent = append(clippedContent, cnt)
+// 				clippedContent[snpt] = struct{}{}
+// 			}
+// 		}
+
+// 	}
+
+// 	var allContent string
+// 	for k := range clippedContent {
+// 		// allContent = append(allContent, k)
+// 		allContent += "\n" + k
+// 	}
+
+// 	return allContent, nil
+// }
+
+func extractSnippet(lines []string, startLine, endLine, startChar, endChar int32) string {
+	if startLine > endLine || startLine < 0 || endLine >= int32(len(lines)) {
+		return ""
+	}
+
+	result := make([]string, 0)
+
+	for i := startLine; i <= endLine; i++ {
+		line := lines[i]
+		if startChar == 0 && endChar == 0 {
+			result = append(result, line)
+			continue
+		}
+
+		if i == startLine {
+			if startChar < 0 || startChar >= int32(len(line)) {
+				return ""
+			}
+
+			if i == endLine {
+				if endChar < startChar || endChar > int32(len(line)) {
+					return ""
+				}
+				result = append(result, line[startChar:endChar])
+			} else {
+				result = append(result, line[startChar:])
+			}
+		} else if i == endLine {
+			if endChar < 0 || endChar > int32(len(line)) {
+				return ""
+			}
+			result = append(result, line[:endChar])
+		} else {
+			result = append(result, line)
+		}
+	}
+
+	return strings.Join(result, "\n")
```

Comment:
```suggestion
	if startLine > endLine || startLine < 0 || endLine >= int32(len(lines)) {
		return ""
	}

	c := make([]string, endLine-startLine+1)
	copy(c, lines[startLine:endLine+1])
	n := len(c) - 1
	c[0] = c[0][startChar:]
	c[n] = c[n][:endChar]

	return strings.Join(c, "\n")
```

This is a bit terser and I think handles an end condition that spot-checking this implementation seems to miss.